src/dowel/logger.py

```diff
@@ -213,6 +214,48 @@ def log(self, data):
                     type(data).__name__))
             self._warn(warning)
 
+    def reorganize_file(self,data,file_name):
+        """After some adjustments for TabularInput,
+        we can leave the cell blank, if the value of the key is missing
+        
+        Thus, next goal is to expand header
+        ,and expand old rows with empty cells for the new key
+        
+        First, we have recorded all different keys in TabularInput.headers.
+        We will use comma , seperate them ,then form the header of csv file.
+
+        Second, because the header sequence of output of example csv file,
+         "progress.csv", is inverted order, I keep the pattern and fill each row
+         with adding comma. The number of adding comma equals 
+         # of header's comma - # of this row' comma.
+         This step achieve expanding old rows with empty cells for the new key.
+
+         Finally, we copy the temp file to the target file.
+
+        || Most of the code are necessary but messy string or file operations ||
+        """
+        with open("temp_file.txt",'w+') as ft:
```

Comment:
I think that using a temporary file like this is generally the right approach, but it helps to name it relative to the file it's a "copy" of, so that e.g. multiple calls from different programs don't accidentally use the same `"temp_file.txt"`.